src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let sourceString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let string = this[i];
+
+      if (string === undefined || string === null) {
+        string = '';
+      }
+
+      if (i === (this.length - 1)) {
+        sourceString += `${string}`;
+      } else {
+        sourceString += `${string}${separator}`;
+      }
```

Comment:
```suggestion
      const string = this[i];

      if (string !== undefined && string !== null) {
        sourceString += string;
      }

      if (i !== (this.length - 1)) {
        sourceString += separator;
      }
```
I think you can optimize your solution a bit.
Also, `string` is not the best name, because it's actually just a char